index.js

```diff
@@ -78,38 +70,42 @@ module.exports = class DfinityTx extends Message {
 
   /**
    * deserializes the message and returns a new instance of `DfinityTx`
-   * @param {Buffer} raw - the serialized raw messsage
+   * @param {Buffer} raw - the serialized raw message
    * @return {Promise} resolve with a new instance of `DfinityTx`
    */
   static async deserialize (raw) {
-    const p = new Pipe(raw)
-    const type = p.read(1)
-    assert.equal(type[0], 1, 'txs should start with type 1')
+    const isSigned = raw.length > 65
```

Comment:
how does this work? the data field is arbitary size so the the data field it self could be more then 65 bytes without the message being signed